src/postprocess/band_structure.jl

```diff
@@ -30,39 +30,23 @@ function high_symmetry_kpath(model; kline_density=40)
                 kpath=[["-½", "½"]], kbranches=[1:length(kcoords)])
     end
 
-    # - Brillouin.jl expects the input direct lattice to be in the conventional lattice
-    #   in the convention of the International Table of Crystallography Vol A (ITA).
-    # - spglib uses this convention for the returned conventional lattice,
-    #   so it can be directly used as input to Brillouin.jl
-    # - The output k-Points and reciprocal lattices will be in the CDML convention.
-    conv_latt = get_spglib_lattice(model; to_primitive=false)
-    sgnum     = spglib_spacegroup_number(model)  # Get ITA space-group number
-    direct_basis   = Bravais.DirectBasis(collect(eachcol(conv_latt)))
-    primitive_latt = Bravais.primitivize(direct_basis, Bravais.centering(sgnum, 3))
-
-    primitive_latt ≈ collect(eachcol(model.lattice)) || @warn(
-        "DFTK's model.lattice and Brillouin's primitive lattice do not agree. " *
-        "The kpath selected to plot the band structure might not be most appropriate."
-    )
-
-    kp     = Brillouin.irrfbz_path(sgnum, direct_basis)
+    # TODO: Wrap this into a function? (Repeated pattern in spglib.jl)
+    spg_positions, spg_numbers, _ = spglib_atoms(model.atoms)
+    cell = Spglib.Cell(model.lattice, spg_positions, spg_numbers)
+    dset = Spglib.get_dataset(cell)
+
+    # If the input cell is a supercell (without any distortion), Brillouin.irrfbz_path does not work.
+    if !isapprox(det(cell.lattice), det(dset.primitive_lattice)) # check volumes agree
+        error(DomainError(cell, "`cell` is a supercell and untreatable by `Brillouin.irrfbz_path`. "))
+    end
```

Comment:
The issue is that irrfbz_path in Brillouin.jl will error, not warn. I agree that giving some k path is better than giving nothing.

If the model is not treatable because it is a supercell, I think there are two ways to give a k path: use a the cell same lattice but (1) with maximal symmetry (i.e. one atom per unit cell) or (2) with minimal symmetry.

Reply:
My first rationale would be that if you have a supercell, you sort of look into that cell as one entity, (e.g. because you later want to perturb it and break the symmetry), so I would argue one would want minimal symmetry.

But now that I think about this some more I'm not sure if the best way to go about this is to be automatic. It would be better in such "unusual" cases to force the user to specify what he wants. What I'm thinking is to ask for an explicit list of symmetries to be taken into account when doing the plotting. In practice we already have `model.symmetries` in `model`, which specifies the symmetries the user has set to be used. Ideally one would just use that as a default and allow the user of `high_symmetry_kpath` to overwrite them.

If given those symmetries the algorithm works, it should just go through. Else I think its fine to error and ask the user to manually tinker with the symmetries.

What do you think @jaemolihm?

Reply:
I think that the idea of using the given symmetry is great. But, I cannot think of how to implement it.

To run `irrfbz_path`, one needs 1) the space group, and 2) the standard primitive cell. The space group for given symmetry can be easily found using spglib. But, AFAIK spglib does not have a function to standardize the cell using only given symmetries.

Also, now I am not sure how to implement the minimal symmetry option either because of the same reason. One could construct an artificial cell with the same lattice but many atoms randomly placed to break all symmetries, but this seems a bit too dirty.

Another option is to use take the k path of the primitive unit cell (which will have a large BZ than the supercell) and fold it back to the supercell BZ. A downside for this option is that the high symmetry k point labels lose their meaning.

Reply:
> But, I cannot think of how to implement it.

No worries! I did not expect this to be easily doable ... I was just hypothesising about the "best case". The implementation we had before we went to Bravais has been way less correct / useful in any case, so we are certainly moving in the right direction with this PR :smile:.

> But, AFAIK spglib does not have a function to standardize the cell using only given symmetries.

I think you can do that, but it probably involves doing some nasty calls deep into the library, which will not be fun ... or easy to do. Probably best to put this on the long list of #533.

> Another option for solution, not discusseded in my previous comment, is to use take the k path of the primitive unit cell (which will have a large BZ than the supercell) and fold it back to the supercell BZ. A downside for this option is that the high symmetry k point labels lose their meaning.

Actually all things considered, I think this is short-term the best solution. Great suggestion!

Reply:
Hi, I made a PR to Brillouin.jl https://github.com/thchr/Brillouin.jl/pull/16 that implements the alternative I mentioned (taking k path of the primitive cell). If that gets merged, I will update this PR.

Reply:
Now, `Brillouin.irrfbz_kpath`, as well as `DFTK.high_symmetry_kpath` gives the primitive cell k path for undistorted supercells.